core/src/main/java/org/apache/gravitino/storage/relational/service/UserMetaService.java

```diff
@@ -246,6 +249,38 @@ public <E extends Entity & HasIdentifier> UserEntity updateUser(
     return newEntity;
   }
 
+  public List<UserEntity> listUsersByNamespace(Namespace namespace, List<Field> skippingFields) {
+    AuthorizationUtils.checkUserNamespace(namespace);
+    String metalakeName = namespace.level(0);
+
+    if (skippingFields.contains(UserEntity.ROLE_IDS)
+        && skippingFields.contains(UserEntity.ROLE_NAMES)) {
```

Comment:
It seems that `skippingFields` is not flexible enough, you still need to explicitly check these fields and execute by different branches.

Reply:
Maybe I can use the annotation to optimize this code.  I can update a code to let you see.

Reply:
After considering twice, I decide an interface `SupportsSkippingFields` to express the `list user namespace` handler. I also wrap the logic which we select the handler according to skipping fields into `SupportsSKippingFieldsHandlers`.